turtleFSI/modules/newtonsolver.py

```diff
@@ -64,20 +64,26 @@ def newtonsolver(F, J_nonlinear, A_pre, A, b, bcs, lmbda, recompute, recompute_t
         if recompute_for_timestep or recompute_frequency or recompute_residual or recompute_initialize:
             if MPI.rank(MPI.comm_world) == 0 and verbose:
                 print("Compute Jacobian matrix")
+            # Assemble non-linear part of Jacobian, keep sparsity pattern (keep_diagonal=True)
             assemble(J_nonlinear, tensor=A,
                          form_compiler_parameters=compiler_parameters,
                          keep_diagonal=True)
+            # Add non-linear and linear part of Jacobian
             A.axpy(1.0, A_pre, True)
+            # Insert ones on diagonal to make sure the matrix is non-singular (related to solid pressure being zero)
             A.ident_zeros()
             [bc.apply(A) for bc in bcs]
 
-        # Compute right hand side
-        b = assemble(-F, tensor=b)
+        # Aseemble right hand side vector
+        b = assemble(-F)
 
-        # Apply boundary conditions and solve
+        # Apply boundary conditions before solve
         [bc.apply(b, dvp_["n"].vector()) for bc in bcs]
+        # Solve the linear system A * x = b where A is the Jacobian matrix, x is the Newton increment and b is the residual
```

Comment:
It is the `- residual`, so you are solving $dF/du[\delta u] = - F(u_k)$, `u_{k+1} = u_k + \delta u$